### PR TITLE
CAW-34: Update dependencies and tests

### PR DESCRIPTION
**Pull request by roblafeve**
## Description for [34](https://github.com/cremalab/crema-app-web/issues/34)

This updates all dependencies, includes `@testing-library/react` and updates tests to use it, adds a "type" generator, and various cleanup items

### 📋 Acceptance Criteria

> Checked off by the PR **Assignee(s)**

- [x] Update deps
- [x] Include `@testing-library/react`
- [x] Add generator for types
- [x] Cleanup

### 👩‍🔬 Test Instructions

1. Run `npm ci`
1. Run generators and confirm they are working as expected

## 🔎 Reviewer Checklist

> Checked off by the PR **Reviewers**

### Required

> These always need to be checked

- [x] Merge destination is correct
- [x] Code is correct as understood and conforms to quality standards
- [x] Tests have been added where appropriate (unit, visual, end-to-end)
- [x] Acceptance Criteria have been met

---

>### Roles & Responsibilities
>
>#### 👨‍💻 Assignee
>
>- Initiator of this PR (be sure to set in GitHub UI)
>- Addresses feedback and change requests
>- Merges PR once approved (usually deletes branch unless `develop` or `release`)
>
>#### 👩‍💻 Reviewer
>
>- Invited to review PR by **Assignee** (via GitHub UI)
>- Is expected to complete a review and address followup

### CODE REVIEW

File: src/components/App/test.tsx [hunk 1]
```diff
@@ -1,17 +1,19 @@
...
     const div = document.createElement("div")
     ReactDOM.render(<App />, div)
     ReactDOM.unmountComponentAtNode(div)
   })
 
   it("renders expected structure", () => {
-    const received = shallowRender(<App />)
-    expect(received).toMatchSnapshot()
+    const { container } = render(<App />)
+    expect(container).toMatchSnapshot()
```
Comment (blaketarter) [inline src/components/App/test.tsx]: @roblafeve Does it make sense now that we have removed default snapshot tests to still include one as an example, or should we change this over to a test similar to the generated tests?

File: src/index.css [hunk 2]
```diff
@@ -1,3 +1,11 @@
+html { 
+  box-sizing: border-box; 
+} 
+
+html *, html *:before, html *:after {
```
Comment (blaketarter) [inline src/index.css:5]: 🎉

File: src/components/App/test.tsx (hunk 1 above)
Comment (roblafeve) [reply src/components/App/test.tsx]: @blaketarter good question—I'm not sure 😬

**Review by blaketarter — APPROVED** [review]
:shipit:

**Pull request merged**